csharp/src/Google.Protobuf.Test/CodedInputStreamAsyncTest.cs

```diff
@@ -0,0 +1,502 @@
+#region Copyright notice and license
```

Comment:
I do not have a very good idea. How about some help methods which take an isAsync option to call CodedInputStream/CodedOutputStream APIs. For example :
uint ReadRawVarint32(CodedInputStream input, bool isAsync)
{
    return isAsync ? (await input.ReadRawVarint32Async(CancellationToken.None)) : input.ReadRawVarint32();
}

Does this work? 
For C++ we can use macro, I think C# does not support macro?